### PR TITLE
Traditional JavaScript Array sort & sorting tests

### LINKED ISSUE #21: Documentation on comparator is wrong
"If you pass a function that names two incoming arguments it will be used as a native Array.prototype.sort, where you get passed two models and return a 1, 0, -1 to specify how they compare."

Double checked in the source and I did not see any way this could be functioning. A quick attempt to utilize this also failed.

### PR DESCRIPTION
**Pull request by herkyl**
Adds the traditional JavaScript array sort: function(a, b).
Adds tests for sorting.

This fixes #21 and #11. Also helps with #20.

### CODE REVIEW

**Comment by kamilogorek** [comment]
Looks good! Could you please update other collection-related repos as well please?

**Comment by wraithgar** [comment]
This would also remove the lodash.sortBy requirement, can you remove it from package.json and remove the require statement?

File: ampersand-filtered-subcollection.js [hunk 1]
```diff
@@ -157,10 +157,15 @@ assign(FilteredCollection.prototype, Events, {
...
     },
 
-    _sortModels: function (newModels) {
-        var comparator = this.comparator || this.collection.comparator;
+    _sortModels: function (newModels, comparator) {
+        comparator = comparator || this.comparator || this.collection.comparator;
         if (comparator) {
-            newModels = sortBy(newModels, comparator);
+            if (typeof comparator === 'string' || comparator.length === 1) {
+                newModels = sortBy(newModels, comparator);
```
Comment (lukekarrys) [inline ampersand-filtered-subcollection.js]: @wraithgar Looks like `sortBy` is still used here so I don't think it can be removed.

Comment (kamilogorek) [inline ampersand-filtered-subcollection.js]: That's right, it's still used for key based sorting.

Comment (herkyl) [inline ampersand-filtered-subcollection.js]: @kamilogorek @lukekarrys is correct. `sortBy` is still used when the comparator is a string or a function with 1 argument

**Comment by herkyl** [comment]
@kamilogorek Which repos are you talking about exactly? AmpersandJS/ampersand-collection already does this, however with a much more verbose approach since they don't use `sortBy` at all https://github.com/AmpersandJS/ampersand-collection/blob/master/ampersand-collection.js#L211

**Comment by kamilogorek** [comment]
@herkyl `ampersand-subcollection` still uses `sortBy` https://github.com/AmpersandJS/ampersand-subcollection/blob/master/ampersand-subcollection.js#L175

**Comment by herkyl** [comment]
@kamilogorek I thought `ampersand-filtered-subcollection` was the successor to `ampersand-subcollection`. `ampersand-subcollection` is deprecated according to the readme and it will not accept updates.

**Comment by kamilogorek** [comment]
@herkyl this is what happens when you travel too much... Thanks for pointing that out :)

**Comment by herkyl** [comment]
What is the state with this? Also could someone give me feedback on #22 so I could provide a pull request.

**Comment by lukekarrys** [comment]
I'm a +1 on this. @kamilogorek was that a +1 from you as well? If so I think we can merge this b

File: ampersand-filtered-subcollection.js (hunk 1 above)
Comment (wraithgar) [inline ampersand-filtered-subcollection.js]: Ah yep missed that somehow. carry on.

**Comment by kamilogorek** [comment]
@lukekarrys that's right, +1 from me as well

**Comment by lukekarrys** [comment]
Thanks @herkyl!

**Comment by lukekarrys** [comment]
Published as `v2.0.3`

**Comment by herkyl** [comment]
Thanks for accepting!

**Pull request merged**